walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,70 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Set;
+import java.util.List;
+import java.util.ArrayList;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    List<TimeRange> times = new ArrayList<>();
+    if (request.getDuration() > 1440) {
+      return times;
+    }
+    //asume all time is valid
+    times.add(TimeRange.WHOLE_DAY);
+    for (Event event : events) {
+      if (hasOverlappingAttendendees(event.getAttendees(), request.getAttendees())) {
+        TimeRange eventTime = event.getWhen();
+        for (int i = 0; i < times.size(); i++) {
```

Comment:
hmm, it seems strange to use a for loop like this if you are going to decrement indices like this. Maybe this be better represented using a queue?

Reply:
do you mean to change the times object to a queue object instead?
or to use a queue and a list object together so I don't have to decrement indices when removing things?

Reply:
I think just a queue will be fine, since you just need to handle processing one TimeRange at a time (dequeue) and add newly created TimeRange back (enqueue), and the process would be over once that queue is empty. But you also need another data structure to hold the final time ranges (I think this corresponds to when you call "times.set", since the queue will be empty once finished. Essentially, have two queues, one Queue "times" and one Queue "adjustedTimes". Enqueue all of the TimeRange that are done processing into "adjustedTimes" and then when "times" is empty, have "times = adjustedTimes" so you have the correct Queue for the next iteration. Something similar can be done for the for loop where you remove time slots, but you would only need to return the final outcome of "adjustedTimes", since there is no more iteration. I think this should work, but let me know if something doesn't make sense. Thanks!

Reply:
That makes sense. I change the implementation to that. Thanks for the explanation!

Reply:
Changed to a queue implementation!